core/src/main/java/org/apache/iceberg/TrackedDataFileAdapter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Adapter that wraps a TrackedFile and presents it as a DataFile.
+ *
+ * <p>This adapter allows TrackedFile instances to be used in contexts that expect DataFile. The
+ * adapter returns null for partition data and column-level statistics until ContentStats is
+ * implemented.
+ *
+ * <p>TODO: When ContentStats is implemented, use spec to extract partition from contentStats.
+ */
+class TrackedDataFileAdapter implements DataFile {
+  private final TrackedFile<?> trackedFile;
+
+  @SuppressWarnings("UnusedVariable")
+  private final PartitionSpec spec;
+
+  TrackedDataFileAdapter(TrackedFile<?> trackedFile, PartitionSpec spec) {
+    if (trackedFile.contentType() != FileContent.DATA) {
+      throw new IllegalStateException(
+          "Cannot convert TrackedFile with content type "
+              + trackedFile.contentType()
+              + " to DataFile");
+    }
+    this.trackedFile = trackedFile;
+    this.spec = spec;
+  }
+
+  @Override
+  public String manifestLocation() {
+    return trackedFile.manifestLocation();
```

Comment:
Using an inner class would allow you to reference the private variables directly.